app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -786,6 +789,62 @@ class Settings(private val appContext: Context) : PreferencesHolder {
             0
         )
 
+    private val numTimesFeedbackPromptShown: Int
+        get() = preferences.getInt(
+            appContext.getPreferenceKey(R.string.pref_key_feedback_prompt_shown),
+            0
+        )
+
+    fun incrementNumTimesFeedbackPromptShown() {
+        preferences.edit().putInt(
+            appContext.getPreferenceKey(R.string.pref_key_feedback_prompt_shown),
+            numTimesFeedbackPromptShown + 1
+        ).apply()
+    }
+
+    private val numTimesOpenedAfterInstall: Int
+        get() = preferences.getInt(
+            appContext.getPreferenceKey(R.string.pref_key_times_opened_after_install),
+            0
+        )
+
+    fun incrementNumTimesOpenedAfterInstall() {
+        preferences.edit().putInt(
+            appContext.getPreferenceKey(R.string.pref_key_times_opened_after_install),
+            numTimesOpenedAfterInstall + 1
+        ).apply()
+    }
+
+    private val timeWhenPromptWasLastShown: Int
+        get() = preferences.getInt(
+            appContext.getPreferenceKey(R.string.pref_key_time_prompt_shown),
+            0
+        )
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun incrementTimeWhenPromptWasLastShown() {
+        preferences.edit().putInt(
+            appContext.getPreferenceKey(R.string.pref_key_time_prompt_shown),
+            LocalDate.now().dayOfYear
+        ).apply()
+    }
+
+    /*
+     * User feedback prompt is shown when Firefox is set as the default browser and after the
+     * 5th time the user opened the app. This prompt should only be shown once every four months.
+     */
+    @RequiresApi(Build.VERSION_CODES.O)
+    val shouldShowUserFeedbackPrompt: Boolean =
+        numTimesOpenedAfterInstall >= 5
+                && isDefaultBrowser()
+                && (hasBeenFourMonthsSince(timeWhenPromptWasLastShown))
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun hasBeenFourMonthsSince(timeWhenPromptWasLastShown: Int): Boolean {
+        val numDays = LocalDate.now().dayOfYear - timeWhenPromptWasLastShown
```

Comment:
We might want to account the year into this as well otherwise we might get stuck if gets set late into the year